radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java

```diff
@@ -128,12 +128,39 @@ public Completable toCompletable() {
 
 	private final Ledger ledger;
 
+	private RadixApplicationAPI(
+		RadixIdentity identity,
+		RadixUniverse universe,
+		FeeMapper feeMapper,
+		Ledger ledger,
+		List<ActionToParticlesMapper> actionToParticlesMappers
+	) {
+		this.identity = identity;
+		this.universe = universe;
+
+		// TODO: Utilize class loader to discover and load these modules
+		this.uniquePropertyTranslator = new UniquePropertyTranslator();
+		this.messageActionStore = new ActionStore<>(ledger.getAtomStore(), new AtomToDecryptedMessageMapper(universe));
+		this.tokenBalanceStore = new ApplicationStore<>(ledger.getParticleStore(), new TokenBalanceReducer());
+		this.tokenTransferActionStore = new ActionStore<>(ledger.getAtomStore(), new AtomToTokenTransfersMapper(universe));
+		this.tokenStore = new ApplicationStore<>(ledger.getParticleStore(), new TokenReducer());
```

Comment:
Nitpicking, but would it make sense to move these instantiations to the static creator method instead? Usually you'd prefer to return quickly from the constructor, which can't really be guaranteed right now. And also because testing (mocking)...

Reply:
Yes, this definitely needs to be refactored out. But I would rather have tech debt here to remind us that this does need to be figured out rather than prematurely refactoring. I think we still need a few more apps written on top of the api to inform us the correct way to refactor this.